src/Mono.Android/Android.Runtime/AndroidEnvironment.cs

```diff
@@ -355,6 +355,11 @@ static IWebProxy GetDefaultProxy ()
 			return Activator.CreateInstance (handlerType);
 		}
 
+		internal static bool VSAndroidDesignerIsEnabled { get; } = InitializeVSAndroidDesignerIsEnabled ();
+
+		static bool InitializeVSAndroidDesignerIsEnabled () =>
+		    !AppContext.TryGetSwitch ("Java.Interop.TypeManager.VSAndroidDesignerIsEnabled", out bool isEnabled) || isEnabled;
```

Comment:
Naming quibble? Should this instead be `Xamarin.Android.AndroidDesigner.IsSupported`?

Quasi-rationale:

  * This is specific to Xamarin.Android, not Java.Interop, so a `Java.Interop` prefix doesn't quite make sense.
  * While this is used by `TypeManager`, it's no longer *limited* to the TypeManager, so I'm not sure that belongs.
  * A [quick perusal of `dotnet/runtime`](https://github.com/dotnet/runtime/search?q=AppContext+TryGetSwitch&type=code) shows existing switch names such as `System.Diagnostics.Tracing.EventSource.IsSupported`, `System.StartupHookProvider.IsSupported`, `System.Resources.UseSystemResourceKeys`, `System.Diagnostics.Tracing.EventSource.IsSupported`, etc. Thus suffixes of `Supported` or `Enabled`, or prefixes of `Use`, appear to be "common".

Reply:
Ah, I missed that one out, when renaming. Changed it to use `IsSupported` suffix.